components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/servlet/IWAServlet.java

```diff
@@ -49,8 +50,15 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
 
     @Override
     protected void doPost(HttpServletRequest request, HttpServletResponse response)
-            throws ServletException, IOException {
-        String commonAuthURL = IdentityUtil.getServerURL(IWAConstants.COMMON_AUTH_EP, false, true);
+            throws ServletException, IOException, RuntimeException {
+
+        String commonAuthURL;
+        try {
+            commonAuthURL =
+                    ServiceURLBuilder.create().addPath(IWAConstants.COMMON_AUTH_EP).build().getAbsolutePublicURL();
+        } catch (URLBuilderException e) {
+            throw new RuntimeException("Error occurred while building URL in tenant qualified mode.", e);
```

Comment:
Any reason to throw a RuntimeException?

Reply:
We have been generally throwing runtime exceptions when building the URLs and it should be the case since we are building the URLs dynamically using parameters from runtime.